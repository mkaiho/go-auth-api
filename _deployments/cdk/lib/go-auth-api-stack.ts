import * as cdk from 'aws-cdk-lib';
import { Tags } from 'aws-cdk-lib';
import { Subnet, Vpc } from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

interface StageContext {
  name: string;
}

export class GoAuthApiStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const env: string = this.node.tryGetContext("env");
    const context: StageContext = this.node.tryGetContext(env);
    const revision = require("child_process")
      .execSync("git rev-parse HEAD")
      .toString()
      .trim();

    /**
     * VPC
     */
    const vpc = new Vpc(this, `${context.name}-vpc`, {
      vpcName: `${context.name}-vpc`,
      enableDnsHostnames: true,
      enableDnsSupport: true,
      natGateways: 0,
      subnetConfiguration: [],
      cidr: "10.0.0.0/16",
    });

    /**
     * Subnet
     */
    const appPublicSubnet = new Subnet(this, `${context.name}-app-public-subnet`, {
      vpcId: vpc.vpcId,
      cidrBlock: "10.0.0.0/24",
      availabilityZone: "ap-northeast-1a",
      mapPublicIpOnLaunch: false,
    });
    Tags.of(appPublicSubnet).add('Name', `${context.name}-app-public-subnet`)
    const apPrivateSubnet = new Subnet(this, `${context.name}-app-private-subnet`, {
      vpcId: vpc.vpcId,
      cidrBlock: "10.0.1.0/24",
      availabilityZone: "ap-northeast-1a",
      mapPublicIpOnLaunch: false,
    });
    Tags.of(apPrivateSubnet).add('Name', `${context.name}-app-private-subnet`)

    /**
     * ECR
     */
    const image = this.synthesizer.addDockerImageAsset({
      sourceHash: revision,
      directoryName: `${__dirname}/../../../`,
    });
  }
}
