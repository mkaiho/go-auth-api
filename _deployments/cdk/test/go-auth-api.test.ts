import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { GoAuthApiStack as TestStack } from '../lib/go-auth-api-stack';
import { context } from '../cdk.json';

// example test. To run these tests, uncomment this file along with the
// example resource in lib/go-auth-api-stack.ts
test('snapshot test', () => {
    const env = "stage"
    const app = new cdk.App({ context: { env, [env]: context } });
    // WHEN
    const stack = new TestStack(app, 'TestStack');
    // THEN
    const template = Template.fromStack(stack);

    expect(template).toMatchSnapshot();
});
