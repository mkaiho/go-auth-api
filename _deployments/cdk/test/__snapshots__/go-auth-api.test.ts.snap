// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "ecsTaskExecutionRolePolicyTestStackecsTaskExecutionRoleAB894092E9C7AA41": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "ecr:BatchCheckLayerAvailability",
                "ecr:BatchGetImage",
                "ecr:GetDownloadUrlForLayer",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":ecr:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":repository/",
                    {
                      "Fn::Sub": "cdk-hnb659fds-container-assets-\${AWS::AccountId}-\${AWS::Region}",
                    },
                  ],
                ],
              },
            },
            {
              "Action": "ecr:GetAuthorizationToken",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "goauthapilog92D7EFB8",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PolicyTestStackecsTaskExecutionRoleAB894092",
        "Roles": [
          "ecsTaskExecutionRole",
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "goauthapialb7417EE95": {
      "DependsOn": [
        "goauthapiapppublicsubnet1aDefaultRoute062449F5",
        "goauthapiapppublicsubnet1aRouteTableAssociation2B6018D6",
        "goauthapiapppublicsubnet1cDefaultRouteCB680EBF",
        "goauthapiapppublicsubnet1cRouteTableAssociationA0868B6A",
      ],
      "Properties": {
        "LoadBalancerAttributes": [
          {
            "Key": "deletion_protection.enabled",
            "Value": "false",
          },
        ],
        "Name": "go-auth-api-alb",
        "Scheme": "internet-facing",
        "SecurityGroups": [
          {
            "Fn::GetAtt": [
              "goauthapialbsgF17A1428",
              "GroupId",
            ],
          },
        ],
        "Subnets": [
          {
            "Ref": "goauthapiapppublicsubnet1aSubnet35BBFA9E",
          },
          {
            "Ref": "goauthapiapppublicsubnet1cSubnet13708802",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "goauthapialbgoauthapialbhttplistener5449924C": {
      "Properties": {
        "DefaultActions": [
          {
            "TargetGroupArn": {
              "Ref": "goauthapialbtargetgroupD6F4F6F0",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": {
          "Ref": "goauthapialb7417EE95",
        },
        "Port": 80,
        "Protocol": "HTTP",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "goauthapialbsgF17A1428": {
      "Properties": {
        "GroupDescription": "TestStack/go-auth-api-alb-sg",
        "GroupName": "go-auth-api-alb-sg",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "from 0.0.0.0/0:80",
            "FromPort": 80,
            "IpProtocol": "tcp",
            "ToPort": 80,
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "goauthapialbtargetgroupD6F4F6F0": {
      "Properties": {
        "HealthCheckIntervalSeconds": 60,
        "HealthCheckPath": "/health",
        "Matcher": {
          "HttpCode": "200",
        },
        "Name": "go-auth-api-alb-target-group",
        "Port": 3000,
        "Protocol": "HTTP",
        "TargetGroupAttributes": [
          {
            "Key": "stickiness.enabled",
            "Value": "false",
          },
        ],
        "TargetType": "ip",
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "goauthapiappprivatesubnetRouteTable58DD87C2": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-app-private-subnet",
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "goauthapiappprivatesubnetRouteTableAssociation02E87075": {
      "Properties": {
        "RouteTableId": {
          "Ref": "goauthapiappprivatesubnetRouteTable58DD87C2",
        },
        "SubnetId": {
          "Ref": "goauthapiappprivatesubnetSubnet262F9611",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "goauthapiappprivatesubnetSubnet262F9611": {
      "Properties": {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "10.0.11.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-app-private-subnet",
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "goauthapiapppublicsubnet1aDefaultRoute062449F5": {
      "DependsOn": [
        "goauthapiigwattachment",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "goauthapiigw",
        },
        "RouteTableId": {
          "Ref": "goauthapiapppublicsubnet1aRouteTable324B777C",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "goauthapiapppublicsubnet1aRouteTable324B777C": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-app-public-subnet-1a",
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "goauthapiapppublicsubnet1aRouteTableAssociation2B6018D6": {
      "Properties": {
        "RouteTableId": {
          "Ref": "goauthapiapppublicsubnet1aRouteTable324B777C",
        },
        "SubnetId": {
          "Ref": "goauthapiapppublicsubnet1aSubnet35BBFA9E",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "goauthapiapppublicsubnet1aSubnet35BBFA9E": {
      "Properties": {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "10.0.1.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-app-public-subnet-1a",
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "goauthapiapppublicsubnet1cDefaultRouteCB680EBF": {
      "DependsOn": [
        "goauthapiigwattachment",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "goauthapiigw",
        },
        "RouteTableId": {
          "Ref": "goauthapiapppublicsubnet1cRouteTable33262C05",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "goauthapiapppublicsubnet1cRouteTable33262C05": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-app-public-subnet-1c",
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "goauthapiapppublicsubnet1cRouteTableAssociationA0868B6A": {
      "Properties": {
        "RouteTableId": {
          "Ref": "goauthapiapppublicsubnet1cRouteTable33262C05",
        },
        "SubnetId": {
          "Ref": "goauthapiapppublicsubnet1cSubnet13708802",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "goauthapiapppublicsubnet1cSubnet13708802": {
      "Properties": {
        "AvailabilityZone": "ap-northeast-1c",
        "CidrBlock": "10.0.2.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-app-public-subnet-1c",
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "goauthapiclusterA00E79D9": {
      "Properties": {
        "ClusterName": "go-auth-api-cluster",
      },
      "Type": "AWS::ECS::Cluster",
    },
    "goauthapiigw": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-igw",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "goauthapiigwattachment": {
      "Properties": {
        "InternetGatewayId": {
          "Ref": "goauthapiigw",
        },
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "goauthapilog92D7EFB8": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "LogGroupName": "/aws/cdk/ecs-alb-fargate-service/web",
        "RetentionInDays": 1,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "goauthapiserviceServiceBF0B3AF3": {
      "DependsOn": [
        "goauthapialbgoauthapialbhttplistener5449924C",
      ],
      "Properties": {
        "Cluster": {
          "Ref": "goauthapiclusterA00E79D9",
        },
        "DeploymentConfiguration": {
          "MaximumPercent": 200,
          "MinimumHealthyPercent": 50,
        },
        "EnableECSManagedTags": false,
        "HealthCheckGracePeriodSeconds": 60,
        "LaunchType": "FARGATE",
        "LoadBalancers": [
          {
            "ContainerName": "go-auth-api",
            "ContainerPort": 3000,
            "TargetGroupArn": {
              "Ref": "goauthapialbtargetgroupD6F4F6F0",
            },
          },
        ],
        "NetworkConfiguration": {
          "AwsvpcConfiguration": {
            "AssignPublicIp": "DISABLED",
            "SecurityGroups": [
              {
                "Fn::GetAtt": [
                  "goauthapiservicesgC5DDA15E",
                  "GroupId",
                ],
              },
            ],
            "Subnets": [
              {
                "Ref": "goauthapiappprivatesubnetSubnet262F9611",
              },
            ],
          },
        },
        "ServiceName": "go-auth-api-service",
        "TaskDefinition": {
          "Ref": "goauthapitask973C00CB",
        },
      },
      "Type": "AWS::ECS::Service",
    },
    "goauthapiservicesgC5DDA15E": {
      "Properties": {
        "GroupDescription": "TestStack/go-auth-api-service-sg",
        "GroupName": "go-auth-api-service-sg",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": [
          {
            "CidrIp": {
              "Fn::GetAtt": [
                "goauthapivpc70352260",
                "CidrBlock",
              ],
            },
            "Description": {
              "Fn::Join": [
                "",
                [
                  "from ",
                  {
                    "Fn::GetAtt": [
                      "goauthapivpc70352260",
                      "CidrBlock",
                    ],
                  },
                  ":443",
                ],
              ],
            },
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "goauthapiservicesgfromTestStackgoauthapialbsg86DB1AB53000D2F9394B": {
      "Properties": {
        "Description": "Allow alb access",
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "goauthapiservicesgC5DDA15E",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "goauthapialbsgF17A1428",
            "GroupId",
          ],
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "goauthapitask973C00CB": {
      "Properties": {
        "ContainerDefinitions": [
          {
            "Command": [
              "cmd/auth-api-server",
            ],
            "Essential": true,
            "Image": {
              "Fn::Join": [
                "",
                [
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ".dkr.ecr.",
                  {
                    "Ref": "AWS::Region",
                  },
                  ".",
                  {
                    "Ref": "AWS::URLSuffix",
                  },
                  "/",
                  {
                    "Fn::Sub": "cdk-hnb659fds-container-assets-\${AWS::AccountId}-\${AWS::Region}",
                  },
                  ":cacd9973f1151f392b2eb329c107dbded012ce8c",
                ],
              ],
            },
            "LogConfiguration": {
              "LogDriver": "awslogs",
              "Options": {
                "awslogs-group": {
                  "Ref": "goauthapilog92D7EFB8",
                },
                "awslogs-region": {
                  "Ref": "AWS::Region",
                },
                "awslogs-stream-prefix": "ecs",
              },
            },
            "Name": "go-auth-api",
            "PortMappings": [
              {
                "AppProtocol": "http",
                "ContainerPort": 3000,
                "HostPort": 3000,
                "Name": "http-port-mapping",
                "Protocol": "tcp",
              },
            ],
          },
        ],
        "Cpu": "256",
        "ExecutionRoleArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:iam::",
              {
                "Ref": "AWS::AccountId",
              },
              ":role/ecsTaskExecutionRole",
            ],
          ],
        },
        "Family": "go-auth-api-task",
        "Memory": "512",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": [
          "FARGATE",
        ],
        "TaskRoleArn": {
          "Fn::GetAtt": [
            "goauthapitaskTaskRole4F553A1F",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "goauthapitaskTaskRole4F553A1F": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "goauthapivpc70352260": {
      "Properties": {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Name",
            "Value": "go-auth-api-vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "goauthapivpceecrapi50EF525F": {
      "Properties": {
        "PrivateDnsEnabled": true,
        "SecurityGroupIds": [
          {
            "Fn::GetAtt": [
              "goauthapiservicesgC5DDA15E",
              "GroupId",
            ],
          },
        ],
        "ServiceName": {
          "Fn::Join": [
            "",
            [
              "com.amazonaws.",
              {
                "Ref": "AWS::Region",
              },
              ".ecr.api",
            ],
          ],
        },
        "SubnetIds": [
          {
            "Ref": "goauthapiappprivatesubnetSubnet262F9611",
          },
        ],
        "VpcEndpointType": "Interface",
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::VPCEndpoint",
    },
    "goauthapivpceecrdkr34864192": {
      "Properties": {
        "PrivateDnsEnabled": true,
        "SecurityGroupIds": [
          {
            "Fn::GetAtt": [
              "goauthapiservicesgC5DDA15E",
              "GroupId",
            ],
          },
        ],
        "ServiceName": {
          "Fn::Join": [
            "",
            [
              "com.amazonaws.",
              {
                "Ref": "AWS::Region",
              },
              ".ecr.dkr",
            ],
          ],
        },
        "SubnetIds": [
          {
            "Ref": "goauthapiappprivatesubnetSubnet262F9611",
          },
        ],
        "VpcEndpointType": "Interface",
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::VPCEndpoint",
    },
    "goauthapivpcelogs51747E4F": {
      "Properties": {
        "PrivateDnsEnabled": true,
        "SecurityGroupIds": [
          {
            "Fn::GetAtt": [
              "goauthapiservicesgC5DDA15E",
              "GroupId",
            ],
          },
        ],
        "ServiceName": {
          "Fn::Join": [
            "",
            [
              "com.amazonaws.",
              {
                "Ref": "AWS::Region",
              },
              ".logs",
            ],
          ],
        },
        "SubnetIds": [
          {
            "Ref": "goauthapiappprivatesubnetSubnet262F9611",
          },
        ],
        "VpcEndpointType": "Interface",
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::VPCEndpoint",
    },
    "goauthapivpces39B399B6F": {
      "Properties": {
        "RouteTableIds": [
          {
            "Ref": "goauthapiappprivatesubnetRouteTable58DD87C2",
          },
        ],
        "ServiceName": {
          "Fn::Join": [
            "",
            [
              "com.amazonaws.",
              {
                "Ref": "AWS::Region",
              },
              ".s3",
            ],
          ],
        },
        "VpcEndpointType": "Gateway",
        "VpcId": {
          "Ref": "goauthapivpc70352260",
        },
      },
      "Type": "AWS::EC2::VPCEndpoint",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
